hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -716,81 +708,8 @@ SqlResult execute(CreateTypePlan plan) {
-            }
-
-            final ClassDefinition existingClassDef = getSerializationService(hazelcastInstance).getPortableContext()
-                    .lookupClassDefinition(factoryId, classId, version);
-
-            if (existingClassDef != null) {
-                type = TypesUtils.convertPortableClassToType(plan.name(), existingClassDef, catalog);
-            } else {
-                if (plan.columns().isEmpty()) {
-                    throw QueryException.error("The given FactoryID/ClassID/Version combination not known to the member. " +
-                            "You need to provide column list for this type");
-                }
-
-                type = new Type();
-                type.setName(plan.name());
-                type.setKind(TypeKind.PORTABLE);
-                type.setPortableFactoryId(factoryId);
-                type.setPortableClassId(classId);
-                type.setPortableVersion(version);
-                type.setFields(new ArrayList<>());
-
-                for (int i = 0; i < plan.columns().size(); i++) {
-                    final TypeDefinitionColumn planColumn = plan.columns().get(i);
-                    type.getFields().add(new Type.TypeField(planColumn.name(), planColumn.dataType()));
-                }
-            }
-        } else if (SqlConnector.COMPACT_FORMAT.equals(format)) {
-            if (plan.columns().isEmpty()) {
-                throw QueryException.error("Column list is required to create Compact-based Types");
-            }
-            type = new Type();
-            type.setKind(TypeKind.COMPACT);
-            type.setName(plan.name());
-            final List<Type.TypeField> typeFields = plan.columns().stream()
-                    .map(typeColumn -> new Type.TypeField(typeColumn.name(), typeColumn.dataType()))
-                    .collect(Collectors.toList());
-            type.setFields(typeFields);
-
-            final String compactTypeName = plan.option(SqlConnector.OPTION_TYPE_COMPACT_TYPE_NAME);
-            if (compactTypeName == null || compactTypeName.isEmpty()) {
-                throw QueryException.error("Compact Type Name must not be empty for Compact-based Types.");
-            }
-            type.setCompactTypeName(compactTypeName);
-        } else if (SqlConnector.JAVA_FORMAT.equals(format)) {
-            final Class<?> typeClass;
-            try {
-                typeClass = ReflectionUtils.loadClass(plan.options().get(SqlConnector.OPTION_TYPE_JAVA_CLASS));
-            } catch (Exception e) {
-                throw QueryException.error("Unable to load class: '"
-                        + plan.options().get(SqlConnector.OPTION_TYPE_JAVA_CLASS) + "'", e);
-            }
-
-            type = TypesUtils.convertJavaClassToType(plan.name(), plan.columns(), typeClass);
-        } else {
-            throw QueryException.error("Unsupported type format: " + format);
-        }
-
+        final Type type = new Type(plan.name(), plan.columns(), plan.options());
```

Comment:
we do not do any validation of options when creating a Type. Maybe we should?
1. if `format` is specified for type also other parameters for that format must be specified
2. will we allow partial overriding of properties, eg. define `OPTION_TYPE_PORTABLE_CLASS_ID` on type level but inherit `OPTION_TYPE_PORTABLE_FACTORY_ID` and `OPTION_TYPE_PORTABLE_CLASS_VERSION` from mapping? This might be tricky to use.

Reply:
1. You cannot specify serialization format for a type. However, `SqlCreateMapping`/`SqlCreateType` only check against duplicate options. Maybe they should be improved and kept in line with `SqlCreateJob`/`SqlAlterJob`?
2. Currently, type metadata is inseparable: `factoryId`, `classId` and `version` must be defined together to be effective; otherwise, `portableId` is assumed to be undefined. I think allowing partial metadata will only cause confusion and not bring any value.

Reply:
As you pointed out, the validation of mapping and type options depend on the connector and serialization format, which seems to be the reason why it was not implemented in `SqlCreateMapping`/`SqlCreateType`. I think, it should be kept as is.